### PR TITLE
Improve media device selection

### PR DESCRIPTION
**Pull request by lukasIO**
depends on https://github.com/livekit/client-sdk-js/pull/759 to get the correct default selection after publishing

### CODE REVIEW

File: packages/react/src/components/controls/MediaDeviceSelect.tsx [hunk 1]
```diff
@@ -87,9 +89,12 @@ export function MediaDeviceSelect({
     }
   }, [onDeviceListChange, devices]);
 
+  React.useEffect(() => {
+    onActiveDeviceChange?.(activeDeviceId);
+  }, [activeDeviceId]);
```
Comment (Ocupe) [inline packages/react/src/components/controls/MediaDeviceSelect.tsx:94]: missing dep `onActiveDeviceChange`?

File: packages/react/src/components/controls/MediaDeviceSelect.tsx [hunk 2]
```diff
@@ -87,9 +89,12 @@ export function MediaDeviceSelect({
...
+
   const handleActiveDeviceChange = async (deviceId: string) => {
-    setActiveMediaDevice(deviceId);
-    onActiveDeviceChange?.(deviceId);
+    await setActiveMediaDevice(deviceId, { exact: exactMatch });
   };
```
Comment (Ocupe) [inline packages/react/src/components/controls/MediaDeviceSelect.tsx:98]: Maybe consider inlining?

File: packages/react/src/components/controls/MediaDeviceSelect.tsx [hunk 3]
```diff
@@ -49,6 +49,7 @@ export interface MediaDeviceSelectProps extends React.HTMLAttributes<HTMLUListEl
   onActiveDeviceChange?: (deviceId: string) => void;
   onDeviceListChange?: (devices: MediaDeviceInfo[]) => void;
   initialSelection?: string;
+  exactMatch?: boolean;
```
Comment (Ocupe) [inline packages/react/src/components/controls/MediaDeviceSelect.tsx:52]: Maybe a short docstring to explain what exactMatch does and what happens if there is no exact match.

**Review by Ocupe — APPROVED** [review]
Only small comments.

File: turbo.json [hunk 4]
```diff
@@ -36,6 +36,7 @@
     },
     "api-check": {
       "dependsOn": ["^build"]
-    }
+    },
+    "api-check:update": {}
```
Comment (Ocupe) [inline turbo.json:40]: I think this is missing 
```
"api-check:update": {
     "outputs": ["temp/*", "ect/*"]
}
```

Comment (lukasIO) [reply turbo.json:40]: what is the outputs section used for? 🤔

File: packages/react/src/components/controls/MediaDeviceSelect.tsx (hunk 2 above)
Comment (lukasIO) [reply packages/react/src/components/controls/MediaDeviceSelect.tsx]: it's now wrapped in a try/catch so leaving as is

File: turbo.json (hunk 4 above)
Comment (lukasIO) [reply turbo.json:40]: I think that's only useful if some other task would depend on `update`, right? 
For that case I don't really see a scenario where we would want to keep that info cached, as a call to it means something went wrong/API changed

**Pull request merged**